File: Classes/PageCell.m
// PageCell.m
//
// Kraftstoff


#import "PageCell.h"
#import "PageViewController.h"
#import "AppDelegate.h"


const CGFloat PageCellDefaultRowHeight = 44.0;


@implementation PageCell


+ (CGFloat)rowHeight
{
    return PageCellDefaultRowHeight;
}


+ (NSString *)reuseIdentifier
{
	return NSStringFromClass (self);
}


- (id)init
{
	if ((self = [super initWithStyle:UITableViewCellStyleDefault reuseIdentifier:[[self class] reuseIdentifier]])) {

		[self finishConstruction];
        self.detailTextLabel.hidden = YES;
	}

	return self;
}


- (void)prepareForReuse
{
    // Suppress default behaviour
}


- (void)finishConstruction
{
    // Overridepoint for subclasses
}


- (void)configureForData:(id)object viewController:(id)viewController tableView:(UITableView *)tableView indexPath:(NSIndexPath *)indexPath
{
    // Overridepoint for subclasses
}

@end

File: Classes/PageCell.swift
//
//  PageCell.swift
//  kraftstoff
//
//  Created by Ingmar Stein on 03.05.15.
//
//

import UIKit

class PageCell: UITableViewCell {
  class var reuseIdentifier: String {
    String(describing: self)
  }

  required init() {
    super.init(style: .default, reuseIdentifier: type(of: self).reuseIdentifier)

    detailTextLabel?.isHidden = true
  }

  @available(*, unavailable)
  required init(coder _: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }

  func configureForData(_: [String: Any], viewController _: UIViewController, tableView _: UITableView, indexPath _: IndexPath) {
    // Overridepoint for subclasses
  }

  func reset() {}
}
